python/pyspark/worker.py

```diff
@@ -1640,9 +1640,20 @@ def check_return_value(res):
             def convert_to_arrow(data: Iterable):
                 data = list(check_return_value(data))
                 if len(data) == 0:
-                    return [
-                        pa.RecordBatch.from_pylist(data, schema=pa.schema(list(arrow_return_type)))
-                    ]
+                    return [pa.RecordBatch.from_pylist(data, schema=pa.schema(list(arrow_return_type)))]
+                
+                # if unnecessary, try to skip expensive LocalDataToArrowConversion
+                needs_conversion = any(
+                    LocalDataToArrowConversion._need_converter(field.dataType, field.nullable)
+                    for field in return_type.fields
+                )
+                
+                if not needs_conversion:
+                    try:
+                        return [pa.RecordBatch.from_pylist(data, schema=pa.schema(list(arrow_return_type)))]
+                    except Exception:
+                        pass
```

Comment:
This seems dangerous. What are some exceptions that can be thrown here? We should propagate these exceptions to users.  It would be great to add some unit tests as well.